### PR TITLE
Added_Parser_Tests

### PR DESCRIPTION
**Pull request by adodrndarr**
Added some Parser tests for the HTTP Methods and Headers...

### CODE REVIEW

File: Server.Tests/ParserTests.cs [hunk 1]
```diff
@@ -89,5 +89,70 @@ public void Request_With_Body_Should_Return_Correct_Body()
...
             Assert.AreEqual("THIS IS THE BODY", actual.Body);
         }
+
+
+        [TestMethod]
+        public void Requests_With_Specific_Method_Should_Return_Correct_Method_Value_Test()
+        {
+            // 1. Arrange
+            RequestParser parser = new RequestParser();
+            var GETrequestString = @"GET /one/two?three=4 HTTP/1.1";
```
Comment (sweko) [inline Server.Tests/ParserTests.cs:99]: These are 6 different tests, so write 6 different methods. A test should have a single mode of failure - what information do you have if `Requests_With_Specific_Method_Should_Return_Correct_Method_Value_Test` reports failure?

File: Server.Tests/ParserTests.cs [hunk 2]
```diff
@@ -89,5 +89,70 @@ public void Request_With_Body_Should_Return_Correct_Body()
...
+            Assert.AreEqual(Method.Delete, actualDelete.Method);
+            Assert.AreEqual(Method.Options, actualOptions.Method);
+            Assert.AreEqual(Method.Head, actualHead.Method);
+        }
+
+        [TestMethod]
+        public void Request_With_Specific_Header_Should_Return_Correct_Header_Value_Test()
+        {
+            // 1. Arrange
+            RequestParser parser = new RequestParser();
```
Comment (sweko) [inline Server.Tests/ParserTests.cs:183]: This one is better, but still borderline. We have one action method, and then we do multiple asserts on the result.

**Review by sweko — APPROVED** [review]

File: Server.Tests/ParserTests.cs [hunk 3]
```diff
@@ -89,5 +89,70 @@ public void Request_With_Body_Should_Return_Correct_Body()
...
+            Assert.AreEqual(Method.Get, actualGet.Method);
+            Assert.AreEqual(Method.Post, actualPost.Method);
+            Assert.AreEqual(Method.Put, actualPut.Method);
+            Assert.AreEqual(Method.Delete, actualDelete.Method);
+            Assert.AreEqual(Method.Options, actualOptions.Method);
+            Assert.AreEqual(Method.Head, actualHead.Method);
+        }
+
+        [TestMethod]
+        public void Request_With_Specific_Header_Should_Return_Correct_Header_Value_Test()
```
Comment (sweko) [inline Server.Tests/ParserTests.cs:180]: The test name is nice, long and correct.

File: Server.Tests/ParserTests.cs (hunk 1 above)
Comment (adodrndarr) [reply Server.Tests/ParserTests.cs:99]: I would probably get information about the first mode of failure that appeared, which might or might not be the only one.
Therefore  I guess and agree that a test should have a single reason to fail most of the time.

Thank you, the changes have been applied.

**Pull request closed without merging**